Extension/src/LanguageServer/client.ts

```diff
@@ -868,6 +877,86 @@ export class DefaultClient implements Client {
+        clients.forEach(client => {
+            if (client instanceof DefaultClient) {
+                const defaultClient: DefaultClient = <DefaultClient>client;
+                defaultClient.configuration.CompilerDefaults = compilerDefaults;
+                defaultClient.configuration.handleConfigurationChange();
+            }
+        });
+    }
+
+    public async handleCompilerQuickPick(): Promise<void> {
+        const settings: OtherSettings = new OtherSettings();
+        const paths: string[] = compilerDefaults.knownCompilers.map(function (a: configs.KnownCompiler): string { return a.path; });
+        paths.push(localize("selectAnotherCompiler.string", "Select another compiler on my machine"));
+        paths.push(localize("installCompiler.string", "Help me install a compiler"));
+        paths.push(localize("noConfig.string", "Do not configure a compiler (not recommended)"));
+        const index: number = await ui.showSelectCompiler(paths);
+        if (index === paths.length - 1) {
+            paths[index] = "";
+        } else if (index === paths.length - 2) {
+            switch (os.platform()) {
+                case 'win32':
+                    vscode.commands.executeCommand('vscode.open', "https://go.microsoft.com/fwlink/?linkid=2217614");
+                    return;
+                case 'darwin':
+                    vscode.commands.executeCommand('vscode.open', "https://go.microsoft.com/fwlink/?linkid=2217706");
+                    return;
+                default: // Linux
+                    vscode.commands.executeCommand('vscode.open', "https://go.microsoft.com/fwlink/?linkid=2217615");
+                    return;
+            }
+        } else if (index === paths.length - 3) {
+            const result: vscode.Uri[] | undefined = await vscode.window.showOpenDialog();
+            if (result !== undefined && result.length > 0) {
+                util.addTrustedCompiler(compilerPaths, result[0].fsPath);
+                settings.defaultCompiler = result[0].fsPath;
+                compilerDefaults = await this.requestCompiler(compilerPaths);
+                this.updateClientConfigurations();
+                return;
+            }
+        } else {
+            util.addTrustedCompiler(compilerPaths, paths[index]);
+        }
+        // If a compiler is selected, update the default.compilerPath user setting.
+        if (index < paths.length - 3) {
+            settings.defaultCompiler = paths[index];
+        }
+        compilerDefaults = await this.requestCompiler(compilerPaths);
+        this.updateClientConfigurations();
+    }
+
+    async promptSelectCompiler(command: boolean): Promise<void> {
+        const selectCompiler: string = localize("selectCompiler.string", "Select Compiler");
+        const confirmCompiler: string = localize("confirmCompiler.string", "Yes");
+        const settings: OtherSettings = new OtherSettings();
+        if (compilerDefaults.compilerPath !== "") {
```

Comment:
This "if/else" should be removed, i.e. it shouldn't ever show the "No compiler detected, please install a compiler." Having no compiler detected just means we didn't detect a compiler in the hardcoded spots we decide to look, so the list of specific compilers will be empty , but the other options in the quickpick should still be valid/shown.

Reply:
It looks like may be we want the "else" still but it should show a prompt that shows the selectCompiler.

Reply:
Oh, I see you added code for that below, but it's at the wrong spot, since it's hitting the 1st else, i.e. compilerDefaults.compilerPath is ``.

Reply:
Also, if the `compilerDefaults.compilerPath` check is bypassed, it ends up showing `The compiler was found on this computer` so something may need to handle that.